symphony-bdk-legacy/sms-sdk-renderer-java/src/test/java/it/services/SmsRendererTest.java

```diff
@@ -0,0 +1,136 @@
+package it.services;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.junit.Test;
+import services.SmsRenderer;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+public class SmsRendererTest {
+
+  @Test
+  public void renderInBotJSONObjectTest() throws IOException {
+    this.testJsonObject(SmsRenderer.SmsTypes.LIST);
+    this.testJsonObject(SmsRenderer.SmsTypes.TABLE);
+    this.testJsonObject(SmsRenderer.SmsTypes.INFORMATION);
+    this.testJsonObject(SmsRenderer.SmsTypes.NOTIFICATION);
+    this.testJsonObject(SmsRenderer.SmsTypes.ALERT);
+    this.testJsonObject(SmsRenderer.SmsTypes.SIMPLE);
+  }
+
+  @Test
+  public void renderInBotJSONArrayTest() throws IOException {
+    this.testJsonArray(SmsRenderer.SmsTypes.LIST);
+    this.testJsonArray(SmsRenderer.SmsTypes.TABLE);
+    this.testJsonArray(SmsRenderer.SmsTypes.INFORMATION);
+    this.testJsonArray(SmsRenderer.SmsTypes.NOTIFICATION);
+    this.testJsonArray(SmsRenderer.SmsTypes.ALERT);
+    this.testJsonArray(SmsRenderer.SmsTypes.SIMPLE);
+  }
+
+  @Test
+  public void renderInBotStringTest() throws IOException {
+    this.testString(SmsRenderer.SmsTypes.LIST);
+    this.testString(SmsRenderer.SmsTypes.TABLE);
+    this.testString(SmsRenderer.SmsTypes.INFORMATION);
+    this.testString(SmsRenderer.SmsTypes.NOTIFICATION);
+    this.testString(SmsRenderer.SmsTypes.ALERT);
+    this.testString(SmsRenderer.SmsTypes.SIMPLE);
+  }
+
+  // Private methods
+  private void testString(final SmsRenderer.SmsTypes smsTypes) throws IOException {
+    assertNotNull(smsTypes);
+
+    final String fileName = smsTypes.getName().toLowerCase() + "Json.json";
+    final String jsonMessageContext = this.getStringFromFile("/SmsRenderer/JSONData/"+fileName);
+
+    final String compiledTemplate = this.minifyHTML(SmsRenderer.renderInBot(jsonMessageContext, smsTypes));
+    assertNotNull(compiledTemplate);
```

Comment:
Here we don't test the content of `compiledTemplate`. It would be much better to test the content, as we would know what exactly is the output of `SmsRenderer` functions.

Reply:
Ok canI do it but I had suggested to just test if `compiledTemplate` is not `null` and you had accepted (in the chat room)